### ISSUE TITLE
Calls can't function properly, keeps occuring "Network Failed"

### ISSUE THREAD

**Issue by njuoy** [signalapp/Signal-Android#13429]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
The issue occurs on all my contacts, I can't use the call function, it will show "Network Failed" in a few second after I trigger call function. However, the contacts are able to call me but with "signal connecting...", and I immediately receive "Missed voice call" notification after the contacts call.

### Steps to reproduce
- Choose a contact to call.
- Press on the Call icon (either voice call or video call, both occurs the same issue).
- The issue "Network Failed" will occur after a few seconds.

**Actual result:** The screen shows "Network Failed" and the contacts will not be notified that someone is calling them as well as won't have missed call notification.
**Expected result:** The screen should show ringing and notify the contacts that someone is calling them.

### Device info

**Device:** HONOR X9A RMO-NX1
**Android version:** 13 (MagicOS 7.1)
**Signal version:** 6.46.7

### Link to debug log

https://debuglogs.org/android/6.46.7/4225c0122815120cb3b68d738f81ed11ad316494a8f35ab82ef7cf304ea3d623

**Comment by cody-signal** [comment]
If you try sending a normal message to the people you can't call does it work to call them afterwards? It appears you may be stuck in a proof required loop due to a bug in handling proof required for call messages.

**Comment by cody-signal** [comment]
Are you using the same VPN on desktop as your phone?

**Comment by njuoy** [comment]
Yes, I used same vpn on desktop. Before trying to link with the signal desktop app, I have tried calling while not on vpn and still show the issue.

**Comment by MoralCode** [comment]
I'm experiencing this issue on LineageOS21 on a google pixel 6a

**Comment by MoralCode** [comment]
Update: i suspect my issue may be different. I was able to fix it by disconnecting from my tailscale VPN.

**Comment by rugk** [comment]
Also have this issue and it's quite strange. Disabling the VPN (rethink dns) fixes the issue.
